### PR TITLE
add picth loss for xiaoice sing

### PR DESCRIPTION
**Pull request by DrWelles**

### CODE REVIEW

File: muskit/svs/xiaoice/pitchloss.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+from muskit.layers.fastspeech.duration_predictor import (
+    DurationPredictorLoss,  # noqa: H301
+)
+from muskit.torch_utils.nets_utils import make_non_pad_mask
+from muskit.torch_utils.nets_utils import make_pad_mask
+
+
+
+
+class PitchLoss(torch.nn.Module):
```
[the remark this answers is not in the document]
Comment (DrWelles) [reply muskit/svs/xiaoice/pitchloss.py:21]: > The pitch loss seems to be very similar, can we merge the pitchloss and vuv loss?

Of course we can. But it may results in more effort to understand the code in my opinion? If you persist , please comment "Yes" and I will merge these losses.

**Comment by ftshijt** [comment]
Please let me know if the PR is ready to review

**Comment by DrWelles** [comment]
> Please let me know if the PR is ready to review

ready to merge. The update of this version includes the uvu and pitch loss and opensionger recipe. The phone annotation and the new singing vocoder are not provided yet.

File: egs/TEMPLATE/svs1/pyscripts/audio/midi_extraction.py [hunk 2]
```diff
@@ -110,51 +161,17 @@ def utt2ref_channels(x, d=utt2ref_channels_dict) -> Tuple[int, ...]:
...
-                    wave, rate = soundfile.read(wavpath, dtype=np.float64)
-                    if wave.ndim == 2 and utt2ref_channels is not None:
-                        wave = wave[:, utt2ref_channels(uttid)]
+        for line in tqdm(fscp):
+            process_wav2midi(line, writer, utt2ref_channels, args)
+    #         pool.apply_async(process_wav2midi, (line, writer, utt2ref_channels, args,))
+    # pool.close()
+    # pool.join()
+    # for p in processes:
+    #     p.get()
```
Comment (ftshijt) [inline egs/TEMPLATE/svs1/pyscripts/audio/midi_extraction.py:173]:
```suggestion
```

File: egs/TEMPLATE/svs1/pyscripts/audio/test_midi_extraction.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
```
Comment (ftshijt) [inline egs/TEMPLATE/svs1/pyscripts/audio/test_midi_extraction.py:1]: We do not need this?

File: egs/TEMPLATE/svs1/scripts/utils/copy_data_dir.sh [hunk 4]
```diff
@@ -48,7 +48,8 @@ export LC_ALL=C
 srcdir=$1
 destdir=$2
 
-echo "${srcdir}"
+# echo "${srcdir}"
+# echo "${destdir}"
```
Comment (ftshijt) [inline egs/TEMPLATE/svs1/scripts/utils/copy_data_dir.sh:52]:
```suggestion
```

File: egs/TEMPLATE/svs1/svs.sh [hunk 5]
```diff
@@ -320,7 +322,7 @@ if ! "${skip_data_prep}"; then
         
         if [ "${feats_type}" = raw ]; then
             log "Stage 2: Format wav.scp: data/ -> ${data_feats}/"
-            for dset in "${train_set}" "${valid_set}" ${test_sets} ; do
+            for dset in ${test_sets} "${valid_set}"   "${train_set}" ; do
```
Comment (ftshijt) [inline egs/TEMPLATE/svs1/svs.sh:325]: why we need such changes?

File: egs/jsut/svs1/local/data.sh [hunk 6]
```diff
@@ -63,7 +63,7 @@ if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
         mv ${src_data}/segments.tmp ${src_data}/segments
         mv ${src_data}/label.tmp ${src_data}/label
         mv ${src_data}/text.tmp ${src_data}/text
-	cat ${src_data}/segments | awk '{printf("%s jsut\n", $1);}' > ${src_data}/utt2spk
+	    cat ${src_data}/segments | awk '{printf("%s jsut\n", $1);}' > ${src_data}/utt2spk
```
Comment (ftshijt) [inline egs/jsut/svs1/local/data.sh:66]:
```suggestion
	cat ${src_data}/segments | awk '{printf("%s jsut\n", $1);}' > ${src_data}/utt2spk
```

File: egs/opencpop/svs1/run.sh [hunk 7]
```diff
@@ -12,8 +12,11 @@ fmax=7600
 n_fft=2048
 n_shift=300
 win_length=1200
+f0min=60
+f0max=1100
 
-score_feats_extract=frame_score_feats   # frame_score_feats | syllable_score_feats
+expdir=exp_f0
```
Comment (ftshijt) [inline egs/opencpop/svs1/run.sh:18]:
```suggestion
```

**Review by ftshijt — COMMENTED** [review]
Minor comments. Please clean the code next time for PR

**Comment by DrWelles** [comment]
Done

**Comment by ftshijt** [comment]
Many thanks! When time permits, it would be great if you can also start a PR to ESPnet with the update~
